### PR TITLE
upgrade github.com/cespare/xxhash/v2 to v2.1.2

### LINKED ISSUE #105133: cespare should update to v2.1.2
**What happened:**

When imported and build with -buildmode=plugin or anything use dynamic link, go will throw

when dynamic linking, R15 is clobbered by a global variable access and is used here:
[cespare:54](https://github.com/cespare/xxhash/issues/54)

and this happened in go 1.17 and 1.17.1
截图

so, the sctive dependency component "cespare" should be update v2.1.2 or later.

**Environment:**
Kubernetes version (use kubectl version): 1.22 and 1.22.1 fail,
Cloud provider or hardware configuration:
OS (e.g: cat /etc/os-release): ubuntu 20.04.2
go：1.17, 1.17.1

**Anything else we need to know?:**

/sig Release

### PR DESCRIPTION
**Pull request by pacoxu**
Fixes  #105133

See https://github.com/cespare/xxhash/compare/v2.1.2...main for more details.
- [Format with Go 1.17 gofmt to use //go:build constraints](https://github.com/cespare/xxhash/commit/532df6ae81bf8c2b37d4a375e13f720d2f28a81f)

/kind bug

### CODE REVIEW

**Comment by liggitt** [comment]
I don't understand why dependency-stats thinks this increased direct deps:

```diff
-	"directDependencies": 162,
+	"directDependencies": 163,
```

@RinkiyaKeDad

**Comment by liggitt** [comment]
> I don't understand why dependency-stats thinks this increased direct deps

oh, I guess we express a version opinion about this now that is newer than the single use in prometheus client_golang. This version should really be bumped in https://github.com/prometheus/client_golang, tested and released before bumping here

**Comment by pacoxu** [comment]
I opened https://github.com/prometheus/client_golang/pull/914 for prometheus client golang.
Not sure if we need upgrade it before or after that is merged.

**Comment by pacoxu** [comment]
github.com/prometheus/client_golang is currently v1.11.0. The PR to update its `cespare` version was merged.

/hold 
to wait for next release of  github.com/prometheus/client_golang

**Comment by dims** [comment]
@pacoxu please rebase!

**Comment by pacoxu** [comment]
> github.com/prometheus/client_golang is currently v1.11.0. The PR to update its `cespare` version was merged.
> 
> /hold to wait for next release of [github.com/prometheus/client_golang](github.com/prometheus/client_golang)

@dims 
I rebased the PR. `directDependencies` would +1 as Jordon pointed out.


If this is merged, we should leave a task to upgrade `github.com/prometheus/client_golang` which needs to wait for a new release of it.

**Comment by pacoxu** [comment]
```
-	"directDependencies": 159,
+	"directDependencies": 160,
```

**Comment by liggitt** [comment]
is there an update to pull in the new version of https://github.com/prometheus/client_golang that references this?

**Comment by pacoxu** [comment]
> is there an update to pull in the new version of https://github.com/prometheus/client_golang that references this?

Still no new version. See https://github.com/prometheus/client_golang/issues/946 😄

**Comment by pacoxu** [comment]
Update the status here: Prometheus client owner will release a new version after https://github.com/prometheus/client_golang/pull/955 is merged.

**Comment by pacoxu** [comment]
/hold cancel
upgrade prometheus/client_golang to v1.12.0(common to v0.32.1)
https://github.com/prometheus/client_golang/releases/tag/v1.12.0

File: go.mod [hunk 1]
```diff
@@ -9,100 +9,353 @@ module k8s.io/kubernetes
 go 1.16
```
Comment (liggitt) [inline go.mod:9]: the drastic changes in this file are unexpected... you might need to reset and redo the pin-dependency.sh / update-vendor.sh changes

Comment (pacoxu) [reply go.mod]: Reset is done.

/skip 
for pull-kubernetes-e2e-aks-engine-windows-containerd

**Comment by kolyshkin** [comment]
@pacoxu I think you need to fix your name in git config. The commits have

> From: -e <paco.xu@daocloud.io>

which does not look right to me.

**Comment by pacoxu** [comment]
@kolyshkin  Thanks for reminding me. It is a misconfiguration on my laptop.

**Pull request merged**